components/keys/src/types.rs

```diff
@@ -149,10 +151,10 @@ impl Key {
     pub fn decode_ts_from(key: &[u8]) -> Result<TimeStamp, codec::Error> {
         let len = key.len();
         if len < number::U64_SIZE {
-            return Err(codec::Error::KeyLength);
+            return Err(codec::ErrorInner::KeyLength.into());
         }
-        let mut ts = &key[len - number::U64_SIZE..];
-        Ok(number::decode_u64_desc(&mut ts)?.into())
+        let ts = &key[len - number::U64_SIZE..];
+        Ok(NumberCodec::decode_u64_desc(&ts).into())
```

Comment:
Prefer to use the `read_xxx` interface like: `let ts = key.read_u64_desc()?.into();`.

Reply:
done